File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitTimeSheet/EngineerVisitTimeSheetValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace EngineerVisitTimeSheets
    {
        // make sure that contact object is valid
        public class EngineerVisitTimeSheetValidator : BaseValidator
        {
            public void Validate(EngineerVisitTimeSheet oEngineerVisitTimeSheet)
            {

                // make sure that contact object is valid
                if (oEngineerVisitTimeSheet.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oEngineerVisitTimeSheet.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitTimeSheet/EngineerVisitTimeSheetValidator.vb
Namespace Entity

Namespace EngineerVisitTimeSheets

Public Class EngineerVisitTimeSheetValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oEngineerVisitTimeSheet As EngineerVisitTimeSheet)

        'make sure that contact object is valid
        If oEngineerVisitTimeSheet.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oEngineerVisitTimeSheet.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

        
	
        If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
            Throw New ValidationException(Me)
        End If

    End Sub

End Class

End Namespace

End Namespace
